server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -1078,7 +1078,17 @@ public static ClusterBlock createUUIDBasedBlock(ClusterBlock clusterBlock) {
                 state = allocationService.reroute(state, "indices opened");
 
                 for (OpenIndicesTask task : tasks) {
-                    builder.success(task, new LegacyClusterTaskResultActionListener(task, currentState));
+                    builder.success(task, new ActionListener<>() {
+                        @Override
+                        public void onResponse(ClusterState clusterState) {
+                            task.onPublicationComplete();
```

Comment:
I think we don't need to do anything here (except maybe comment why we're not doing anything here)

```suggestion
                            // listener is notified at the end of acking
```